homeassistant/components/icloud/account.py

```diff
@@ -110,9 +115,36 @@ def setup(self) -> None:
                 self._icloud_dir.path,
                 with_family=self._with_family,
             )
-        except PyiCloudFailedLoginException as error:
+        except PyiCloudFailedLoginException:
             self.api = None
-            _LOGGER.error("Error logging into iCloud Service: %s", error)
+            # Login failed which means credentials need to be updated.
+            username = self._config_entry.data[CONF_USERNAME]
+            message = (
```

Comment:
We shouldn't add this. Instead we should add something generic to Home Assistant whenever it sees a config flow on the `reauth` step.

Reply:
Are you referring to the notification I am creating? I can remove it, I agree that a more generic handling of reauth notifications would be good.